drafts/str4d-overwinter-activation/draft1.rst

```diff
@@ -0,0 +1,196 @@
+===========
+
+The key words "MUST", "MUST NOT", "SHOULD", and "MAY" in this document are to be interpreted as described in
+RFC 2119.
+
+
+Abstract
+========
+
+This proposal defines a mechanism for coordinating upgrades of the Zcash network, in order to remove ambiguity
+about when network upgrades will activate, provide defined periods in which users should upgrade their local
+software, and minimize the risks to both the upgrading network and those users opting out of the changes.
+
+
+Motivation
+==========
+
+Zcash is a *consensual currency*: nobody is ever going to force someone to use a specific version or a
+specific branch of Zcash. [#consensual-currency]_ As such, different sub-communities will always have the
+freedom to choose different variants or branches which offer different design trade-offs.
+
+The current Zcash software includes an *auto-senescence* feature, causing nodes running a particular version
+to automatically shut down 16 weeks after that version was released. This was implemented for several reasons
+[#release-lifecycle]_:
+
+- It gives the same systemic advantage of removing old software as auto-upgrade behavior.
+
+- It requires users to individually choose one of the following options:
+
+  - Upgrade to a more recent software release from the main network.
+
+  - Upgrade to an alternative release.
+
+  - Ignore the "deprecation" and keep running the older software.
+
+Developers can rely on this cadence for coordinating network upgrades. Once the last pre-upgrade software
+version has been deprecated, they can reasonably assume that all node operators on the network either support
+the upgraded rules, or have explicitly chosen not to follow them.
+
+However, this behaviour is not sufficient for performing network upgrades. A globally-understood on-chain
+activation mechanism is necessary so that nodes can unambiguously know at what point the changes from an
+upgrade come into effect (and can enforce consensus rule changes, for example).
+
+
+Specification
+=============
+
+The following constants are defined for every network upgrade:
+
+BRANCH_ID
+  A globally-unique 32-bit identifier.
+
+ACTIVATION_HEIGHT
+  The block height at which the network upgrade rules will come into effect, and be enforced as part of the
+  blockchain consensus.
+
+  For removal of ambiguity, the block at height ``ACTIVATION_HEIGHT - 1`` is subject to the pre-upgrade
+  consensus rules, and would be the last common block in the event of a persistent pre-upgrade branch.
+
+  It MUST be greater than the value of ``DEPRECATION_HEIGHT`` in the last software version that will not
+  contain support for the network upgrade. It SHOULD be chosen to be reached approximately three months after
+  the first software version containing support for the network upgrade is released.
+
+Activation mechanism
+--------------------
+
+There MUST only be one network upgrade at a time in progress. Concretely, this means that the Zcash blockchain
```

Comment:
Only one network upgrade at a time... perhaps clarify this is for a specific client implementation X on a specific branch moving to a new branch, right?  Multiple cloners could branch around the same time and that would still be safe for client X moving from branch A to branch B.